SSINS/sky_subtract.py

```diff
@@ -114,28 +114,64 @@ def diff(self):
         attributes are also adjusted so that the resulting SS object passes
         UVData.check()
         """
-
+        # want to reorder to baseline order
+        # baseline order guarantees that the baseline time axis is oriented so that we can safely assume where indices for diffing are
+        # by looking at the blt axis array
         if self.blt_order != 'baseline':
             warnings.warn("Reordering data array to baseline order to perform differencing.")
             self.reorder_blts(order='baseline')
 
-        # index accumulator
-        ind_acc = 0
+        # basline index: indicates which bl (in bl order) is being processed
+        bl_num = 0
+
         # Just treat every baseline independently - make no assumptions other than what is guaranteed by baseline ordered objects
+
+        # the baseline array is the baseline for each member of the baseline-time array, so given it is in baseline form it will look like
+        # [ bl1, bl1, bl1, bl1 ... bl1, bl2, bl2, bl2, bl2 ... bl2, bl3 ... bl(n) ]
+        # so differencing it will give zero for most entries and some value where there is a boundary (which we set to 1)
+        testarray = (self.baseline_array[1:] - self.baseline_array[:-1])
+
+        bltaxisboundaries = np.nonzero(testarray)
+        bltaxisboundaries = np.append(bltaxisboundaries, np.size(testarray, 0)) # add last value to end of array (diff won't be able to catch this)
+        bltaxisboundaries = bltaxisboundaries + 1                               # offset entries to correct distance between 0th index and 1st
+        bltaxisboundaries = np.insert(bltaxisboundaries, 0, 0)                  # add 0 to beginning of array (diff won't be able to catch this)
+
+        #need a second array representing what the output's shape will look like -- it's smaller by Nblts because of diffing
+        diffed_baseline_array = self.baseline_array
+        for bl in np.unique(self.baseline_array):
+            index = (np.argwhere(diffed_baseline_array == bl)[0:1])
+            diffed_baseline_array = np.delete(diffed_baseline_array, index, 0)
```

Comment:
Is there some reason you are not just grabbing all the indices at once and feeding `delete` an array for its second argument, rather than explicitly writing a for loop? I'm fine with it. Just want to check my thinking.

Reply:
I don't believe there is a specific reason, but I do think a `for` loop is necessary to accumulate the first index for each unique `baseline_array` value anyways (which might be slightly faster, as `np.delete` would only be called once). I would need to know exactly what `argwhere()`'s returns as its first value is foreach unique baseline in the baseline array since I can't assume that all the baselines show up for a specific number of times.